### DISCUSSION TITLE
Did I install ARM native or not?

### DISCUSSION THREAD

**Discussion by hein-j** [Homebrew/discussions#563] [Getting started] [answered]
Hello,
I just installed Homebrew on my Mac with a M1 chip using the Rosetta translation layer as recommended. 
Then, I successfully installed Git and Node through it.
Now, I'm finding that the commands `brew`, `git`, and `node` work just fine in my terminal _without_ Rosetta.
Did Homebrew recognize that my computer had a M1 chip and do an ARM native install? Then, is Git and Node also built as native ARM executables on my computer?
or...
Is my computer just recognizing Homebrew, Git, and Node as programs needing Rosetta and running them through it without my awareness?
or...?!
Thank you for your time. I'm trying to see if I should do an ARM-native installation when it's officially supported, or if I had somehow stumbled into one already.

**Comment by carlocab** [comment, marked as answer]
`brew config` will tell you the answer. If you don't know how to interpret the output, please paste it here. (And please use markdown to make sure it renders properly as code.)

**Reply by hein-j** [reply]
Hello,

Between my initial post and now, I uninstalled Homebrew because I suspected it was running through Rosetta. I thought this because I went in and looked at the INSTALL_RECEIPT.json of the Git and Node packages that I installed and saw that the value of "arch" was something like "x86_64" (I don't remember the exact value. But it wasn't the "arm64" that it is now, after I reinstalled Homebrew into opt without Rosetta and reinstalled the two packages).

Nevertheless, I would like to know how I could have used` brew config` to find out. Here is the output from the current install, which I'm sure is ARM native. How would it have been different if it was the Intel version? 
I had checked it with the old install, and it had the same CPU, MacOS, and Rosetta2 values as below, so I had thought, "Hmm.. maybe it's the ARM native..." but couldn't be sure.
(I hope I properly applied markdown, but please excuse me--and educate me--if I did not.)

```
HOMEBREW_VERSION: 2.7.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: bca4804a9e48de5319383d3eddadaa7f054c77da
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 6ceb8365ef90b05c3dbaa496d077849dc50bd09c
Core tap last commit: 31 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.30.0 => /opt/homebrew/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.0.1-arm64
CLT: 12.3.0.0.1.1607026830
Xcode: N/A
Rosetta 2: false

```

**Reply by carlocab** [reply]
The things you want to be looking at to determine if you installed software non-natively are `HOMEBREW_PREFIX`, `CPU`, and `Rosetta 2`. The latter two are easy to interpret--I believe you'll have `penryn` instead of `arm_firestorm_icestorm` when in Rosetta here, but I'm a little hazy on the details. 

If you have `HOMEBREW_PREFIX: /usr/local`, then you almost certainly installed things non-natively, and conversely for `HOMEBREW_PREFIX: /opt/homebrew`. If it's neither of the two then things get more complicated, as I think things could change depending on whether you ran `brew install` and `brew config` in Rosetta or not (and you're not always guaranteed to have run both commands in the same mode).

Another way to determine whether you've installed a piece of software natively is using `file /path/to/binary/executable`. For example, on my Intel machine:
```
❯ file /usr/local/bin/python3
/usr/local/bin/python3: Mach-O 64-bit executable x86_64
❯ file /usr/bin/python3
/usr/bin/python3: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64e:Mach-O 64-bit executable arm64e]
/usr/bin/python3 (for architecture x86_64):     Mach-O 64-bit executable x86_64
/usr/bin/python3 (for architecture arm64e):     Mach-O 64-bit executable arm64e
```

>I hope I properly applied markdown

You did great. 👍

**Reply by hein-j** [reply]
I see! The` HOMEBREW_PREFIX `for the previous install was indeed `/usr/local,` so I guess it was a non-native install. The `CPU` was the same (`octa-core 64-bit arm_firestorm_icestorm`), but I do recall seeing `penryn` somewhere... 

The file command seems like a super useful tool to know. That's what I'm going for next time I need to know the binary of something.

Case closed--Thank you for all your help!
